app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java

```diff
@@ -102,22 +97,21 @@ private void drawQRCode() {
      * {@inheritDoc}
      */
     @Override
-    public void onActivityResult(final int requestCode, final int resultCode, final Intent intent) {
+    public void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
         LOGGER.info("Activity resulted");
-        final IntentResult scanResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, intent);
+        final IntentResult scanResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
         if (scanResult == null) {
             LOGGER.error("No scan result in QR Scanner");
         } else if (resultCode == RESULT_OK) {
             final String result = scanResult.getContents();
             LOGGER.info("Adding new contact to database");
             //Name is the first part
-            final String name = result.split(SEMI_COLON)[0];
+            final String name = COMPILE_SEMICOLON.split(result)[0];
             //Key is the second part
-            final IKey key = QRGenerator.deconstructToKey(result.split(SEMI_COLON)[1]);
+            final IKey key = QRGenerator.deconstructToKey(COMPILE_SEMICOLON.split(result)[1]);
```

Comment:
Joost didn't write this (only changed it), but I'll answer because I did. Good point, it is certainly more efficient to just split it once.